practitioner-service/src/test/java/uk/ac/ox/ndph/mts/practitioner_service/validation/RoleAssignmentValidationTests.java

```diff
@@ -1,20 +1,41 @@
 package uk.ac.ox.ndph.mts.practitioner_service.validation;
 
+import okhttp3.mockwebserver.MockWebServer;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.converter.ConvertWith;
 import org.junit.jupiter.params.provider.CsvSource;
 import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.web.reactive.function.client.WebClient;
 import uk.ac.ox.ndph.mts.practitioner_service.NullableConverter;
+import uk.ac.ox.ndph.mts.practitioner_service.client.WebFluxRoleServiceClient;
+import uk.ac.ox.ndph.mts.practitioner_service.exception.RestException;
 import uk.ac.ox.ndph.mts.practitioner_service.model.RoleAssignment;
+
+import java.net.HttpURLConnection;
+
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+import static uk.ac.ox.ndph.mts.practitioner_service.client.WebFluxRoleServiceClientTest.queueErrorResponse;
+import static uk.ac.ox.ndph.mts.practitioner_service.client.WebFluxRoleServiceClientTest.queueRoleResponse;
 
 @ExtendWith(MockitoExtension.class)
 class RoleAssignmentValidationTests {
 
+    private MockWebServer mockBackEnd;
+
+    private RoleAssignmentValidation validator;
+
+    @BeforeEach void initEach() throws Exception {
```

Comment:
Each means that this runs before each test but looks like running it before All can also be fine, can't it?

Reply:
The MockWebServer docs say it's not safe to reuse, so you have to recreate it each time, and that can change the URL that the client talks to.

Reply:
I haven't seen this, can you add a link?

Reply:
I misread - it says that MockWebServer is lightweight enough that you **can** recreate for each test, but not that you have to. However, there is no way of removing requests from the queue, so there is a chance for setup for one test bleeding into another one. I'll change it if you like.